app/src/common/shared/com/igalia/wolvic/utils/ConnectivityReceiver.java

```diff
@@ -21,34 +20,76 @@ public interface Delegate {
         void OnConnectivityChanged(boolean connected);
     }
 
-    private Context mContext;
-    private List<Delegate> mListeners;
+    private static class NetworkWatcher extends LiveData<Boolean> {
```

Comment:
My main doubt about this change is that I think we're duplicating stuff. On the one hand we have the connectivityManager receiving broadcast events, and then we have the NetworkCallback object doing the same and trying to adapt to the legacy code.

I'd remove this patch from the series and think carefully about what to do here.